libkbfs/folder_block_ops.go

```diff
@@ -1800,14 +1813,33 @@ func (fbo *folderBlockOps) ClearCacheInfo(lState *lockState, file path) error {
 // include all the blocks from before the error, except for those that
 // have encountered recoverable block errors themselves.
 func (fbo *folderBlockOps) revertSyncInfoAfterRecoverableError(
-	blocksToRemove []BlockPointer, si, savedSi *syncInfo) {
+	blocksToRemove []BlockPointer, result fileSyncState) {
+	si := result.si
+	savedSi := result.savedSi
+
 	// Save the blocks we need to clean up on the next attempt.
 	toClean := si.toCleanIfUnused
 
+	newIndirect := make(map[BlockPointer]bool)
+	for _, ptr := range result.newIndirectFileBlockPtrs {
+		newIndirect[ptr] = true
+	}
+
+	// Propagate all unrefs forward, except those that belong to new
+	// blocks that were created during the sync.
+	unrefs := si.unrefs
+	for i, unref := range si.unrefs {
+		if newIndirect[unref.BlockPointer] {
+			unrefs = append(unrefs[:1], unrefs[i+1:]...)
```

Comment:
You mean `[:i]`, right?

Even so, I'm not sure this works -- since `unref` is pointing to the same buffer as `si.unrefs`, what happens with the range when you resize `unref`? I'd have to think / look up the spec to make sure that it doesn't cause an entry to be skipped. It's probably clearer just to append to a new array.